### PR TITLE
Added options to bzOptions

### PR DESCRIPTION
**Pull request by DmitrySokolov**
Added new options:
- **hiddenColumns** - an option to hide unneeded columns on the board 
- **defaultProduct**, **defaultMilestone** - default Product and Milestone to load on startup
- **predefinedX** - predefined option values to accelerate page loading, and/or to restrict option values to some set

### CODE REVIEW

**Comment by leif81** [comment]
Thank-you for the contribution @DmitrySokolov ! Been busy but I hope to review it this week.

File: bzkanban.js [hunk 1]
```diff
@@ -108,6 +122,9 @@ function initNav(callback) {
     ],
     function(err, results) {
         console.log("Nav initialized!");
+        if (isLoggedIn() && bzProduct !== "" && bzProductMilestone !== "") {
```
Comment (DmitrySokolov) [inline bzkanban.js:125]: I am not sure - may be this is redundant.

**Comment by leif81** [comment]
Hi @DmitrySokolov, thanks again for the PR. I am just getting around to looking at this pull request more closely.

I have some questions about the changes.

1. `hiddenColumns`

Could you explain the use case for this option?

If there's a bug state you don't want to use, an option you have is to remove or disable the status in the Bugzilla server configuration itself. Would that work instead?

2. `defaultProduct`, `defaultMilestone`

This might be a good thing to add. Are you aware there's a way to have a default product or milestone right now by setting them in the URL. e.g. `index.html?product=Bugzilla&milestone=Bugzilla%206.0`. That way you can bookmark it yourself or share the link with others. Does that meet your needs?

2. `predefinedX`

Could you explain the use cases for this option too? Is it primarily for speeding up page load time, how much time does it save?

Could you explain why restricting values is important? Because if you want to limit access to certain products you can do that in Bugzilla by configuring your user permissions. Or if it's because you want to hide certain components, versions, etc you could also do that in the Bugzilla server itself by disabling those field values. Would any of those options work for you? My preference would to be rely on the configuration of the Bugzilla server itself.

**Comment by DmitrySokolov** [comment]
Hi @leif81,

As for me, Kanban board is not a full featured replacement of BZ, it is only a tool to visualize a state of development. 

So,

1. `hiddenColumns` - some bug states are required by the development process, but they should not be displayed on a board, to keep it concise.

2. `defaultProduct`, `defaultMilestone` - everybody knows the URL `server.com/board/`, but it would be hard to oblige all devs save and update their bookmarks on every milestone change.

3. `predefinedX` - it is for speeding up page load. Depending on a network, we have ~2x, ~3x times.

**Comment by leif81** [comment]
@DmitrySokolov thank-you for this PR. I've reviewed the PR and your comments again. I still have concerns and I'm not ready to merge this in. Here are my latest thoughts.

1. `hiddenColumns`. In my mind this option exposes a risk of misconfiguration if someone doesn't understand bz workflow constraints work. You probably do, but others probably don't realize you can't just hide a column on the board and expect workflow to work. If this is very important to you I ask if you could provide more detail. If you could share, what are your bug states and what state (column) are you hiding?

3. predefinedX - I understand the use case of speeding of page loading. Sounds good. However I would prefer this implemented differently. Hard coded values carry risk of errors and create work for the admin to keep sync'd. I would be interested in a different implementation that uses Web Storage API. i.e. Once the board is loaded from server, it caches the values using Web Storage API, and the future board loads it from local cache using Web Storage API.
 

P.S. I would rather these three changes be split into 3 PR's. If you're still intersted in these changes please reopen a new PR per change.

**Pull request closed without merging**